app/policies/application_policy.rb

```diff
@@ -54,7 +54,7 @@ def author_in_programming_group?
       possible_programming_group = @record.contributor
 
     elsif @record.respond_to? :context # e.g. file
-      possible_programming_group = @record.context.submission.contributor
+      possible_programming_group = @record.context.contributor
```

Comment:
@kiragrammel I changed this line, since the previous solution did not work correctly for me. Please verify whether this change is okay or let's discuss it otherwise.